Source/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -21,43 +22,54 @@
 
 #pragma mark - UIAccessibilityElement
 
-@protocol ASAccessibilityElementPositioning
-
-@property (nonatomic, readonly) CGRect accessibilityFrame;
-
-@end
-
-typedef NSComparisonResult (^SortAccessibilityElementsComparator)(id<ASAccessibilityElementPositioning>, id<ASAccessibilityElementPositioning>);
+static ASSortAccessibilityElementsComparator _userDefinedComparator = nil;
+void setUserDefinedAccessibilitySortComparator(ASSortAccessibilityElementsComparator userDefinedComparator) {
+  _userDefinedComparator = userDefinedComparator;
+}
 
 /// Sort accessiblity elements first by y and than by x origin.
-static void SortAccessibilityElements(NSMutableArray *elements)
+void SortAccessibilityElements(NSMutableArray *elements)
 {
   ASDisplayNodeCAssertNotNil(elements, @"Should pass in a NSMutableArray");
-
-  static SortAccessibilityElementsComparator comparator = nil;
+  
+  static ASSortAccessibilityElementsComparator defaultComparator = nil;
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-      comparator = ^NSComparisonResult(id<ASAccessibilityElementPositioning> a, id<ASAccessibilityElementPositioning> b) {
-        CGPoint originA = a.accessibilityFrame.origin;
-        CGPoint originB = b.accessibilityFrame.origin;
-        if (originA.y == originB.y) {
-          if (originA.x == originB.x) {
-            return NSOrderedSame;
+    defaultComparator = ^NSComparisonResult(NSObject *a, NSObject *b) {
+      CGPoint originA = a.accessibilityFrame.origin;
+      CGPoint originB = b.accessibilityFrame.origin;
+      if (originA.y == originB.y) {
+        if (originA.x == originB.x) {
+          // if we have the same origin, favor shorter items. If heights are the same, favor thinner items. If size is the same ¯\_(ツ)_/¯
+          CGSize sizeA = a.accessibilityFrame.size;
+          CGSize sizeB = b.accessibilityFrame.size;
+          if (sizeA.height == sizeB.height) {
+            if (sizeA.width == sizeB.width) {
+              return NSOrderedSame;
+            }
+            return (sizeA.width < sizeB.width) ? NSOrderedAscending : NSOrderedDescending;
           }
-          return (originA.x < originB.x) ? NSOrderedAscending : NSOrderedDescending;
+          return (sizeA.height < sizeB.height) ? NSOrderedAscending : NSOrderedDescending;
         }
-        return (originA.y < originB.y) ? NSOrderedAscending : NSOrderedDescending;
-      };
+        return (originA.x < originB.x) ? NSOrderedAscending : NSOrderedDescending;
+      }
+      return (originA.y < originB.y) ? NSOrderedAscending : NSOrderedDescending;
+    };
   });
-  [elements sortUsingComparator:comparator];
+  
+  if (_userDefinedComparator) {
+    [elements sortUsingComparator:_userDefinedComparator];
+  } else {
+    [elements sortUsingComparator:defaultComparator];
+  }
```

Comment:
There isn't a way to switch back to the default comparator. You could avoid this conditional by going back to the single `static` variable (`comparator`) that is initially set to the default implementation (using the `dispatch_once` block) but that can be reassigned (customized) via `setAccessibilitySortComparator:`.

Reply:
Couldn't you call `setUserDefinedAccessibilitySortComparator(nil)`? Or do you mean there is no good reason to need to switch? Which I may agree with.

Reply:
`setUserDefinedAccessibilitySortComparator(nil)` could move the single global pointer back to the default implementation.

The core thing I was trying to express is that we could still have a single global `static` pointer pointing at the active comparator rather than conditionally selecting between two pointers.

Reply:
Hmmm... So you are saying we can make `setUserDefinedAccessibilitySortComparator` do something like: 

```
void setUserDefinedAccessibilitySortComparator(ASSortAccessibilityElementsComparator userDefinedComparator) {
  if (userDefinedComparator) {
    comparator = userDefinedComparator;
  } else {
    comparator = ^NSComparisonResult(NSObject *a, NSObject *b) { ... };
  }
)
```

Then in `SortAccessibilityElements` check in the `dispatch_once` that `comparator` isn't nil (in case no one ever called `setUserDefinedAccessibilitySortComparator`) and set it up if it is (I guess by calling `setUserDefinedAccessibilitySortComparator(nil)` which is pretty gross)? 

On the other hand, if we have 1 pointer and `setUserDefinedAccessibilitySortComparator` is called before the first `SortAccessibilityElements` we could overwrite the user's comparator  in the dispatch_once unless we check for a non-nil `comparator`.

Reply:
I'm saying have one `static` pointer named `defaultComparator` (populated by the `dispatch_once` block) and another named `currentComparator`.

```objc
void setAccessibilitySortComparator(ASSortAccessibilityElementsComparator comparator) {
  currentComparator = comparator ?: defaultComparator;
}
```

Reply:
Oh, ok. 
> The core thing I was trying to express is that we could still have a single global static pointer

This was confusing me because it sounded like you had a way to do this with 1 pointer.

Reply:
Oh, I think you're trying to avoid having the default comparator visible outside of the function's scope?